lib/internal/Magento/Framework/Setup/Declaration/Schema/Db/MySQL/DbSchemaReader.php

```diff
@@ -226,6 +238,13 @@ public function readTables($resource)
             )
             ->where('TABLE_SCHEMA = ?', $dbName)
             ->where('TABLE_TYPE = ?', self::MYSQL_TABLE_TYPE);
+
+        if ($tablePrefix = $this->resourceConnection->getTablePrefix()) {
```

Comment:
Could you remove variable creation in the assertion? Like this:
```suggestion
        $tablePrefix = $this->resourceConnection->getTablePrefix();
        if ($tablePrefix) {
```